packages/contracts/deploy/001-ChugSplashManager.ts

```diff
@@ -0,0 +1,93 @@
+/* Imports: External */
+import { DeployFunction } from 'hardhat-deploy/dist/types'
+
+const deployFn: DeployFunction = async (hre) => {
+  const { deployer } = await hre.getNamedAccounts()
+
+  const proxyUpdaterAddress = (await hre.deployments.get('ProxyUpdater'))
+    .address
+
+  // Deploy the ChugSplashManager implementation contract.
+  const { deploy: deployManager } = await hre.deployments.deterministic(
+    'ChugSplashManager',
+    {
+      salt: hre.ethers.utils.solidityKeccak256(
+        ['string'],
+        ['ChugSplashManager']
+      ),
+      from: deployer,
+      log: true,
+    }
+  )
+  const { address: managerAddress, abi: managerAbi } = await deployManager()
+
+  // Get the ChugSplashRegistry's proxy address that *will* be deployed using Create2. We'll need
```

Comment:
Lol, funny little bootstrapping problem going on here. Lemme think for a sec...

Reply:
Honestly the most fun way to do this would be to use ChugSplash to bootstrap itself, but the `ChugSplashManager` relies on the existence of the `ChugSplashRegistry`. Next best thing might be to have a contract called `ChugSplashBootLoader` that does this deployment instead inside of its constructor. Then our entire initial deployment becomes sending a single transaction, which is very ChugSplashy. 

Once the initial system has been deployed, we can manage ChugSplash with ChugSplash!

Reply:
Good idea! Added the ChugSplashBootLoader and took most of the deploy scripts.